tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/behaviours/pageobjects/RunConfigurationSelector.java

```diff
@@ -59,25 +70,225 @@ public PitRunConfiguration getConfiguration(String configName) {
-        boolean runInParallel = bot.checkBox("Mutation tests run in parallel").isChecked();
-        boolean incrementalAnalysis = bot.checkBox("Use incremental analysis").isChecked();
-        String excludedClasses = bot.textWithLabel("Excluded classes (e.g.*IntTest)").getText();
-        String excludedMethods = bot.textWithLabel("Excluded methods (e.g.*toString*)").getText();
-        String avoidCallsTo = bot.textWithLabel("Avoid calls to").getText();
+        String project = bot.textWithLabel(PitArgumentsTab.PROJECT_TEXT).getText();
+        boolean isTestClass = bot.radio(PitArgumentsTab.TEST_CLASS_RADIO_TEXT).isSelected();
+        String testObject;
+        if (isTestClass) {
+            testObject = bot.textWithLabel(PitArgumentsTab.TEST_CLASS_TEXT).getText();
+        } else {
+            testObject = bot.textWithLabel(PitArgumentsTab.TEST_DIR_TEXT).getText();
+        }
+        boolean runInParallel = bot.checkBox(PitPreferences.RUN_IN_PARALLEL_LABEL).isChecked();
+        boolean incrementalAnalysis = bot.checkBox(PitPreferences.INCREMENTAL_ANALYSIS_LABEL).isChecked();
+        String excludedClasses = bot.textWithLabel(PitPreferences.EXCLUDE_CLASSES_LABEL).getText();
+        String excludedMethods = bot.textWithLabel(PitPreferences.EXCLUDE_METHODS_LABEL).getText();
+        String avoidCallsTo = bot.textWithLabel(PitPreferences.AVOID_CALLS_LABEL).getText();
         return new Builder().withName(name).withProjects(project).withRunInParallel(runInParallel)
                 .withIncrementalAnalysis(incrementalAnalysis).withExcludedClasses(excludedClasses)
-                .withExcludedMethods(excludedMethods).withAvoidCallsTo(avoidCallsTo).build();
+                .withExcludedMethods(excludedMethods).withAvoidCallsTo(avoidCallsTo).withTestObject(testObject)
+                .withTestClassOrDir(isTestClass).build();
     }
 
-    private SWTBotTreeItem activateShell() {
-        SWTBotShell shell = bot.shell("Run Configurations");
+    private SWTBotShell activateShell() {
+        // look if shell is already open
+        for (SWTBotShell shell : bot.shells()) {
+            if (shell.getText().equals(RUN_CONFIGURATIONS)) {
+                shell.activate();
+                return shell;
+            }
+        }
+        // shell was not open, open and activate it
+        SWTBotMenuHelper menuHelper = new SWTBotMenuHelper();
+        menuHelper.findMenu(bot.menu(RUN), RUN_CONFIGURATIONS + "...").click();
+        SWTBotShell shell = bot.shell(RUN_CONFIGURATIONS);
         shell.activate();
+        return shell;
+    }
+
+    private SWTBotTreeItem getPitConfigurationItem() {
+        activateShell();
         for (SWTBotTreeItem treeItem : bot.tree().getAllItems()) {
             if ("PIT Mutation Test".equals(treeItem.getText())) {
-                return selectAndExpand(treeItem);
+                return treeItem.select().expand();
             }
         }
         return null;
```

Comment:
This will generate a NPE in the code above that calls getPitConfigurationItem() is that right?
If that's true, maybe an explicit failure instead of returning null might be better and more helpful in case of test failure?

Reply:
I was thinking more of an actual failure, e.g., with `fail` and a detailed message, @echebbi what do you think?